### ISSUE TITLE
RESP3 async pubsub should be owned by `ClientLike`

### ISSUE THREAD

**Issue by feelingsonice** [redis-rs/redis-rs#1338]
The [documented](https://docs.rs/redis/latest/redis/#resp3-async-pubsub) way to subscribe and receive pubsub messages is flawed:

```rust
let client = redis::Client::open("redis://127.0.0.1/?protocol=resp3").unwrap();
let (tx, mut rx) = tokio::sync::mpsc::unbounded_channel();
let config = redis::AsyncConnectionConfig::new().set_push_sender(tx);
let mut con = client.get_multiplexed_async_connection_with_config(&config).await?;
con.subscribe("channel_1").await?;
con.subscribe("channel_2").await?;

loop {
  println!("Received {:?}", rx.recv().await.unwrap());
}
```

The consequence here is that `con` has to be owned and managed by the caller. If not, when the connection is dropped, the `unbounded_channel` is closed. This may sound obvious but it's a big issue if the user just wants to listen to pubsub messages and not do anything else. 

Personally, I have a service whose sole purpose is to listen to pubsub messages, I originally created something like this:

```rust
impl MyTrait for redis::Client {
    async fn subscription(&mut self, topic: &str) -> impl Stream<Item = PushInfo> + Send {
        let (tx, rx) = tokio::sync::mpsc::unbounded_channel();
        self.get_connection_manager_with_config(
            ConnectionManagerConfig::new().set_push_sender(tx),
        )
        .await
        .unwrap()
        .subscribe(topic)
        .await
        .unwrap();

        UnboundedReceiverStream::new(rx)
    }
}
```

But quickly realized that the client was getting dropped right after creation. I think it'd make more sense for the `ClientLike` to provide the receiver for the user. Thoughts?

**Comment by feelingsonice** [comment]
> Sounds like what you want is aio::Pubsub

> do you want the internal connection to be a multiplexed connection, a connection manager, or once its supported, a cluster connection?

I think something like `aio::Pubsub` is what I'm alluding to here. The problem is that I can't access the benefits of the various different connection types when using `aio::Pubsub`. In my case, I wanted a managed connection, but `aio::Pubsub` is strictly created out of `MultiplexedConnection`.

**Comment by nihohit** [comment]
Yes, ATM we don't have a managed pubsub. If that's what you want, see the rest of my comment - you can easily create a struct holding a connection manager and implementing stream.
